atom/browser/native_window.cc

```diff
@@ -805,6 +810,25 @@ void NativeWindow::ScheduleUnresponsiveEvent(int ms) {
       base::TimeDelta::FromMilliseconds(ms));
 }
 
+void NativeWindow::SetHtmlApiFullscreen(bool enter_fullscreen) {
+  // Window is already in fullscreen mode, save the state.
+  if (enter_fullscreen && IsFullscreen()) {
+    forced_fullscreen_ = true;
+    html_fullscreen_ = true;
+    return;
+  }
+
+  // Exit html fullscreen state but not window's fullscreen mode.
+  if (!enter_fullscreen && forced_fullscreen_) {
```

Comment:
I think if we want to exit the html fullscreen we have to enter the html fullscreen first?

Previously entering html fullscreen will be prevented if the window is already in fullscreen, but it has been fixed in this PR.